### PR TITLE
Added TRAIT_DISK_VERIFIER, refactors how fake disk checks work slightly

### PR DESCRIPTION
**Pull request by coiax**
Old system: Fake disk checks on examine() can be done by ghosts, nuke ops and Captains. However, this is
done through isobserver(), assigned_role, and has_antagonist datum.

New system: Check the mob or the mind for the TRAIT_DISK_VERIFIER trait, and allow them
to check the fake that way. Gets rid of stuff being hard coded to "assigned_role".

I think this is a good idea, and was good practice for me doing trait related stuff.

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
what's the benefit of instead of putting the checks on the singular item you now have to sift through various files to ensure a mob does/doesn't have X trait?

**Comment by coiax** [comment]
What do you mean, sift through various files?

I mean, for one thing, means the captain could order a "Nuke Disk Identification pamphlet" for the head of security or something, who could then be taught in the arcane ways of checking serial numbers.

**Comment by ExcessiveUseOfCobblestone** [comment]
for instance if I want to see who can/cannot see le epic examine I have to go to job.dm, dead.dm, and nukeops.dm (and that's assuming I know where to look) vs it all being in the singular file in the singular proc where I can say "Oh just captains, ops, and observers can see the message".

But if you plan on making it malleable then this is fine, just u kno follow thru lol

**Comment by coiax** [comment]
So, you're saying I should put a comment in, explaining who normally has the trait?

**Comment by ExcessiveUseOfCobblestone** [comment]
nah you can easily lookup the trait define it's just a minute pain if I wanted to webbit something :P

File: code/modules/antagonists/nukeop/nukeop.dm [hunk 1]
```diff
@@ -50,6 +50,7 @@
 	. = ..()
 	equip_op()
 	memorize_code()
+	owner.add_trait(TRAIT_DISK_VERIFIER, "nuke-op")
```
Comment (SpaceManiac) [inline code/modules/antagonists/nukeop/nukeop.dm]: Should probably be matched with losing the trait when not a nuke op anymore

File: code/modules/jobs/job_types/captain.dm [hunk 2]
```diff
@@ -50,6 +50,7 @@ Captain
 	duffelbag = /obj/item/storage/backpack/duffelbag/captain
 
 	implants = list(/obj/item/implant/mindshield)
+	mind_traits = list(TRAIT_DISK_VERIFIER)
```
Comment (SpaceManiac) [inline code/modules/jobs/job_types/captain.dm]: Feels iffy to me that this is on the outfit, maybe it should go on the job instead?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is now above zero (1). Feel free to make Feature/Tweak/Balance PRs.

**Pull request merged**